ravenframework/SupervisedLearning/FeatureSelection/RFE.py

```diff
@@ -0,0 +1,815 @@
+        identify highly correlated features to speed up the subsequential search.
+        """
+    spec.addSub(InputData.parameterInputFactory('nFeaturesToSelect',contentType=InputTypes.IntegerType,
+        descr=r"""Exact Number of features to select. If not inputted, ``nFeaturesToSelect'' will be set to $1/2$ """
+        """of the features in the training dataset.""", default=None))
+    spec.addSub(InputData.parameterInputFactory('maxNumberFeatures',contentType=InputTypes.IntegerType,
+        descr=r"""Maximum Number of features to select, the algorithm will """
+        """automatically determine the feature list to minimize a total score.""",
+        default=None))
+    spec.addSub(InputData.parameterInputFactory('onlyOutputScore',contentType=InputTypes.BoolType,
+        descr=r"""If maxNumberFeatures is on, only output score should be"""
+        """considered?.""",
+        default=False))
+    spec.addSub(InputData.parameterInputFactory('searchTol',contentType=InputTypes.FloatType,
+        descr=r"""Relative tolerance for search! Only if maxNumberFeatures is set""",
+        default=1e-4))
+    spec.addSub(InputData.parameterInputFactory('applyClusteringFiltering',contentType=InputTypes.BoolType,
+        descr=r"""Applying clustering correlation  before Augmented-RFE search?""",
+        default=False))
+    spec.addSub(InputData.parameterInputFactory('applyCrossCorrelation',contentType=InputTypes.BoolType,
+        descr=r"""Applying cross correlation in case of subgroupping at the """
+        """end of the Augmented-RFE""",
+        default=False))
+    spec.addSub(InputData.parameterInputFactory('step',contentType=InputTypes.FloatType,
+        descr=r"""If greater than or equal to 1, then step corresponds to the (integer) number
+        of features to remove at each iteration. If within (0.0, 1.0), then step
+        corresponds to the percentage (rounded down) of features to remove at
+        each iteration.""", default=1))
+    subgroup = InputData.parameterInputFactory("subGroup", contentType=InputTypes.InterpretedListType,
+        descr=r"""Subgroup of output variables on which to perform the search. Multiple nodes of this type"""
+        """ can be inputted. The Augmented-RFE search will be then performed on each ``subgroup'' separately and then the"""
+        """ the union of the different feature sets are used for the final ROM.""")
+    spec.addSub(subgroup)
+
+    return spec
+
+  def __init__(self):
+    super().__init__()
+    self.printTag = 'FEATURE SELECTION - Augmented-RFE'
+    self.estimator = None
+    self.nFeaturesToSelect = None
+    self.maxNumberFeatures = None
+    self.searchTol = None
+    self.applyClusteringFiltering = False
+    self.onlyOutputScore = False
+    self.applyCrossCorrelation = False
+    self.step = 1
+    self.subGroups = []
+
+  def __getstate__(self):
```

Comment:
no "setstate" method?

Reply:
yes the getstate is only needed to pop the estimator. If a "retrain" happens the SupervisedLearning "re-sets" the estimator using the ```setEstimator``` method